### PR TITLE
Update release checklist doc

### PR DESCRIPTION
**Pull request by pcholakov**
Minor tidy-up:

- remove SDK release requirement
- add explicit service protocol release requirement

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,10 +108,10 @@ Restate follows [Semantic Versioning](https://semver.org/).
 
 You can safely upgrade from a Restate `x.y` to `x.(y+1)` release without performing any manual data migration, as Restate performs an automatic data migration for you.
 
-To check the compatibility with the SDKs, check the respective READMEs:
+For SDK compatibility, refer to the supported version matrix in the respective READMEs:
 
 * [SDK-Java](https://github.com/restatedev/sdk-java/?tab=readme-ov-file#versions)
-* [SDK-Typescript](https://github.com/restatedev/sdk-typescript/?tab=readme-ov-file#versions)
+* [SDK-TypeScript](https://github.com/restatedev/sdk-typescript/?tab=readme-ov-file#versions)
```
Comment (slinkydeveloper) [inline README.md:114]: Maybe add all the other SDK links too

File: README.md [hunk 2]
```diff
@@ -108,10 +108,10 @@ Restate follows [Semantic Versioning](https://semver.org/).
 
 You can safely upgrade from a Restate `x.y` to `x.(y+1)` release without performing any manual data migration, as Restate performs an automatic data migration for you.
 
-To check the compatibility with the SDKs, check the respective READMEs:
+For SDK compatibility, refer to the supported version matrix in the respective READMEs:
```
Comment (slinkydeveloper) [inline README.md:111]: Perhaps generally, what I would write here is something along the lines of "get in touch with the respective SDK owners to check whether a new release should be made for the SDKs"

File: README.md (hunk 1 above)
Comment (pcholakov) [reply README.md]: Done!

File: README.md (hunk 2 above)
Comment (pcholakov) [reply README.md:111]: Done!

File: docs/dev/release.md [hunk 3]
```diff
@@ -30,6 +30,8 @@ https://github.com/issues?q=is%3Aopen+org%3Arestatedev+label%3Arelease-blocker
 
 Make sure that all [supported service protocol versions](../../crates/types/src/service_protocol.rs) are released.
 
+Confirm if any SDK releases are needed to keep up with the runtime and/or service protocol releases.
```
Comment (slinkydeveloper) [inline docs/dev/release.md:33]: perfect!

**Review by slinkydeveloper — APPROVED** [review]

**Pull request merged**